Integrations/integration-Cymon.yml

```diff
@@ -0,0 +1,543 @@
+description: Analyzes suspicious hashes, URLs, domains and IP addresses
+configuration:
+  - display: Trust any certificate (unsecure)
+    name: unsecure
+    defaultvalue: ""
+    type: 8
+    required: false
+script:
+  script: |
+    ''' IMPORTS '''
+    import datetime
+    import time
+    import requests
+    import json
+    import os
+    import re
+
+    # disable insecure warnings
+    requests.packages.urllib3.disable_warnings()
+
+    if not demisto.params().get('useProxy', False):
+        del os.environ['HTTP_PROXY']
+        del os.environ['HTTPS_PROXY']
+        del os.environ['http_proxy']
+        del os.environ['https_proxy']
+
+    ''' GLOBAL VARS '''
+    SERVER_URL_V1='https://www.cymon.io:443/api/nexus/v1'
+    SERVER_DASHBOARD_URL_V1='https://www.cymon.io:443/api/dashboard/v1'
+    SERVER_URL_V2 = 'https://api.cymon.io/v2/ioc/search'
+
+    USE_SSL = True if demisto.params().get('unsecure') else False
+
+    DEFAULT_HEADERS = {
+      "Content-Type": "application/json"
+    }
+
+    ''' HELPER FUNCTIONS '''
+
+    def http_request(method, url, headers):
+        try:
+            res = requests.request(method,
+                                    url,
+                                    verify=USE_SSL,
+                                    headers=headers)
+
+            if res.status_code == 200:
+                return res.json()
+            # 204 HTTP status code is returned when api rate limit has been exceeded
+            elif res.status_code == 204:
+                return_error("You've reached your API call quota.")
+            elif res.status_code == 404:
+                return {}
+
+            res.raise_for_status()
+
+        except Exception, e:
+            raise(e)
+
+    ''' DOMAIN COMMAND '''
+    def get_domain_full_report(domain):
+        report_results = []
+
+        from_param = 0
+        size_param = 10
+        total = None
+
+        url = '{}/{}/{}?from={}&size={}'.format(SERVER_URL_V2, 'domain', domain, from_param, size_param)
+
+        while total is None or total > from_param:
+            response = http_request('GET', url, DEFAULT_HEADERS)
+
+            hits = response.get('hits', [])
+            for hit in hits:
+                timestamp = datetime.datetime.strptime(hit.get('timestamp'), '%Y-%m-%dT%H:%M:%S.%fZ')
```

Comment:
@MarkShleifer does `strptime` fails on `None`? if so maybe we should adjust `hit.get('timestamp')` in case of an empty timestamp as it doesn't worthy to raise an error in that case.

Reply:
Done